### PR TITLE
Ref/async await error

### LINKED ISSUE #586: improve error handling
- [ ]  add try/catch for async code
- [ ] safely access nested object

### PR DESCRIPTION
**Pull request by huseyincansoylu**
closes #586 

- in progress...

### CODE REVIEW

**Comment by fabriziogianni7** [comment]
@huseyincansoylu is this ready to be reviewed?

File: src/api.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import { config } from "./config"
```
Comment (fabriziogianni7) [inline src/api.ts:1]: @huseyincansoylu this makes sense.
the only thing is that you made this file in this PR but you're not using it.
what I suggest is:
you remove this file from this PR
I can open a new issue to make this file and use it all accross the application

Comment (fabriziogianni7) [reply src/api.ts:1]: also I think this may be done in the github context

Comment (huseyincansoylu) [reply src/api.ts:1]: okay i can do that

**Review by fabriziogianni7 — APPROVED** [review]

**Review by fabriziogianni7 — APPROVED** [review]

**Pull request merged**